chaosaws/__init__.py

```diff
@@ -67,7 +67,29 @@ def aws_client(resource_name: str, configuration: Configuration = None,
         boto3.setup_default_session(
             profile_name=profile_name, region_name=region, **creds)
 
-    return boto3.client(resource_name, region_name=region, **creds)
+    # default config
+    if configuration.get("aws_role_arn") is None:
+        return boto3.client(resource_name, region_name=region, **creds)
+    else:
```

Comment:
Is connecting via 'sts' the only other way?

I don't want to surprise users because they defaulted to sts which wasn't maybe their plan.

I'm not familiar enough with AWS authentication to understand the meaning of such a default here. Could yu please comment?

Reply:
Well they only would use the sts connection if they specify an "aws_role_arn" in the experiment. I could probably add more checks to line 71 incase they input an empty string or something similar. The reason we do this is for a system with multiple AWS accounts, and a central node running experiments. Boto can pull a single role from aws credential files, but you'd have to change them every-time you want to assume a new role. This way, any role passed via the experiment will be assumed dynamically (assuming the original role has those permissions).

Reply:
I certainly like the feature and it makes sense to have it.

As long as the expectations are clear on how the driver should behave.

In this regards, I might suggest:

* more explanation in the CHANGELOG entry
* additional documentation in the README
* Perhaps so more logging (debug level at least) about which option is selected at runtime for auditing of the logs afterwards :)

Cheers!

Reply:
Sounds good. I will make these changes and push them sometime today. Thanks for the quick feedback!

Reply:
Maybe `aws_assume_role_arn` would be a bit more clean on the intent here? I just wonder if it might get confusing if say an action was added that could modify a role and they now have the same key in two places in their experiment configuration. Just a thought

Reply:
Probably a smart call, I'll make the change